### DISCUSSION TITLE
shortcodes in headings?

### DISCUSSION THREAD

**Discussion by d-morrison** [quarto-dev/quarto-cli#10776] [Feature Requests]
### Description

Probably a niche request, but I'm decomposing some homework problems into subfiles, and it would be useful to be able to use the `{{< include >}}` shortcode inside of a heading, like this:

```
---
format: html
---

# heading with insert

## {{< include problem-without-heading-marker.qmd >}}

```

This way, I could reuse the same problem text as a standalone question or as a subpart of a larger question, by setting the heading level in my top-level .qmd.

Here's the current output, for reference:

![image](https://github.com/user-attachments/assets/95a43653-f9ed-4284-9ffb-bafb6ead43c9)

**Comment by cscheid** [comment]
That's not going to be a thing that `include` will ever support - include is documented as supported only at the top level for good reasons we cannot change.

**Reply by d-morrison** [reply]
Fair enough - thanks for the quick answer!

**Comment by mcanouil** [comment]
Why not include your heading in the file included?

That's a bit odd to want to include a whole document as an heading.

**Reply by d-morrison** [reply]
Only the first line of the subfile (`problem-without-heading-marker.qmd` in my example) would be the heading; for example, the subfile might be something like:

```
Derive the MLE

Derive the MLE for an independent sample from a Bernoulli distribution.

::: callout-note

Show your work.

:::
```

Then the full output would be:

![image](https://github.com/user-attachments/assets/c576bfb3-74a3-4150-92a1-5c92c7b8411c)

But I could also use the the same subfile for another assignment where it gets a lower level heading, like this:

```
---
format: html
---

# assignment 2

## Derive MLEs for various distributions

### derive the MLE for the Gaussian distribution.

[some details here]

### {{< include problem-without-heading-marker.qmd >}}

```

I realize it's easy to just exclude the heading text from the subfile and just re-type the heading directly into each main file; I just wondered if it was feasible to make `include` work inline (and I now understand that it's not).
